packages/core/test/actions.test.ts

```diff
@@ -1607,6 +1607,28 @@ describe('choose', () => {
 
     expect(service.state.context).toEqual({ answer: 42 });
   });
+
+  it('Should be able to infer its context types from schema declarations', () => {
+    createMachine({
+      schema: {
+        context: {} as {
+          value: string;
+        }
+      },
+      entry: [
+        choose([
+          {
+            cond: (context) => {
```

Comment:
I'm not entirely sure but I think that this wasn't working even before the typegen PR. 

One of the issues in the codebase is that with this `TContext` is not being inferred properly. I couldn't yet make it work in the codebase but I had some success in inferring it properly in [a slimmed down playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgIILMA9iA8gIwCsIMAeAFQD5kBvAKGWQgA8SB+ALmQAoNmvyASmQBeagDcswACYBuOgF86dMAE8ADigDKoAOYAbCLiiooUOKorURycsgA+tgNoBdeXVCRYiFAFlEABagEADCODDAula0DMgAzggBEAC2cJwxjIwIOJDMYALyjAqFTOBQqunomDgExGRU8kp00iT6cFAoMACuIBjYIMgIHXCQ-onBVrzhkVxjQSCh01FUglySMu4tCG0dyN291QNDECMQ6JMsJFy8YPy2wmLI69KraH01RCRgVu7Hp3PBbj0RgJJKpLjAzLZcAsfK0BTIOBxWiDLA9OEgLrJfDQZAKAA0sQJsQgZVUXD+kHQ3BuzAe1EhWVuADpsuiSgpBITOe4APS85AAWmFyk80HgSGQAIWYRAEV0ACZopDQSk0hDYlkcrCCkSSqSwOV0joQAYjCYzBZSFV+rUvlZKI1lFsdp0eu8jsNRoFgkqqFM5TMpT6ZUs-ZRXs9fl6INKIAqgbFVeCMlDtXkIQikSi2eAuJjsbjiUVCYwDeUuE5NYMY9TafTU5lBizc2ASkUubEXNzBHyBcLBaKYd5JXHZfKAMzKpOJNXpRmomEZ2wc-Vkyoeu0YKcNRRDrwSlA2j51MA7hmxAD6l-IuqaLvaboO-RrJ294wW54D8tmIcWgd0c9IykORmlaR89ndQ5X1OdAv0uBBrj4AQG2eV5jzwT5tx+ZRKVjP8J0TEFZxTBdoVyOEaCzZEaFRdF8yxHEoDxUsWJJMkKVrBBCPrUQLybPhWTRcAOU7HllH5IURQ8YdD2DD9-3lAAWadiLBdVG0XCjdSKNdDQqLgMK3MAVN3JoxRHI9NywkzVOQa9bxXPcVA0FAACFgHEdpgDgcAADEQFM6xU3wTzvN8pAAAlEAAa1pFC1hAxonAAIlCryoB83oIGihAYpStxnXA3Z9g9GD33mCAgu-IMxyWILgI2MDtgg0roLw9BqoQrgPIyrKwAChrDOs08gujN98IUpSiPiEi4A1AT00o6ic2EjFGKLVji1KfTOImzqaT4BsyJbNbRJ7PspMHFzNGQAA5LAAEk5WgILRGQJxyBcT6mDyUlpGRXzVGQNhkAABmQfMIHEaBCsktRboe56YFe6IbDsAAyFEnAAaWQUBkBiiBVCwGBbBcARcZcPF5Hh1z7qel6oDe9HfsgEAAfxpnkC0EGech5AFhhqB3AsuStDAU5x0iABWOzywMt5DmMuWzP3cUfHkyrpd0VXqFhf7kQlqWljlpGmaC-jZvU+dq3InUnN05yxc1oybL11MHLvZQEfcsLMt8gaQA9mxIXS8LstyuLkPuRKmoUVLw4DyPYoKzZiqfMq8LjPWat0X8FJ1vXGtAh8Sqgl8OoQXPuuQXqI6D4vhuV92cLobO-xlmbk3mzT7eXKjERoui80FjbmO27aFb22Dq8O25jrt072SJMTewk-tpLpxHGZRqAADY0Y+r6foNjnAZAYHQYhqHhbhgVfYZ5HoEPqh3sx7G8YJomSbJr7KZxtTYodBt4oHNnvV+wU7Bn05qAPe-NQa81vtAUWslNbG0gDrSBqYFYbhbqeSBTobooAAApQCwOoUguB8TIBxsFPGMDkQ-1JsgXAfNcBUwFkLFBPt6Z1QAofEINDyCygou9MhFDSASMoUI5AKUe4pUoDQlK-cwCKKtowa8LR8BdHzrYURsISg91totJccIABKJAsBQGkKQOIho9A0J6DFEAWAADuIBHR6kUMgLGGDFKRFfgYvIlB1aWSVraGy2DIReyck0R+9dk5B2waHWIScspRVivFWOTwkqKETv7DJOVU6FWaq6SCz4cDlUmpVQRbNDZa2CFg0gIRKCUDzlwEIJd5Bl0zu1LikDuC10Sf1AKkD0IjS+IQ3CMY4z70TJJTRl5tG6MzKxHuC00xmMzEPVa9Ex6FgnltVi09qnoHmbxR4J1mBCRXh2C6G8rqPMfuA6AAB2I+n1vrQL+ufREl8+Y30FtDWGtMH701eVAD5b9WZY1orjLmhNiYsP-s4QBNMQHgp3s-KFR8fns1gdzXmiCuEgpFmE8WktMFLGhVbXBzdImnlpUQl2o4-w62hVoOa9S-lxi5epfWvzOb+I5RQflapUqqIKlbYxAhxWpBZWgyUbsmV2VieQIhCTCmBwCrS96YdtUpzytkoQcdQIJzSoazJeU04Us1ny7lZElpcEsdkGxdiHGmicSAFx7jPHOV6RUrOsy-ycu5YwxpCx5VwHadCH8kaAm6DDQK7pZTWoVyqVXWlQzWCITroaoOtKJn4K+MymZE04xvIWQKJZKy9FUXWXNTZWptnwl2bRVsDFDlsRLOxXaZyEBVsuRo5sNzWznUUOvIAA). The problem is... that even with the proper inference for this generic the type is still not provided correctly contextually for the inner call.

Devan knows some hacks to work around this IIRC but... they are hacks and they involve making our action objects "callable" (by adding a fake call signature to them). 

I think this is a wontfix for now - since fixing this would take a lot of time and might not even be possible at the moment (without the mentioned hacks). We could investigate this in the future - when we have more time to do so.